tests/system/Database/Live/MySQLi/RawSqlTest.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Database\Live\MySQLi;
+
+use CodeIgniter\Database\RawSql;
+use CodeIgniter\Test\CIUnitTestCase;
+use CodeIgniter\Test\DatabaseTestTrait;
+use stdclass;
+use Tests\Support\Database\Seeds\CITestSeeder;
+
+/**
+ * @group DatabaseLive
+ *
+ * @internal
+ */
+final class RawSqlTest extends CIUnitTestCase
+{
+    use DatabaseTestTrait;
+
+    protected $refresh = true;
+    protected $seed    = CITestSeeder::class;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->db->DBDriver !== 'MySQLi') {
+            $this->markTestSkipped('Only MySQLi has its own implementation.');
+        } else {
+            $this->addSqlFunction();
+        }
+    }
+
+    protected function addSqlFunction()
+    {
+        $this->db->query('DROP FUNCTION IF EXISTS setDateTime');
+
+        $sql = "CREATE FUNCTION setDateTime ( setDate varchar(20) )
+                RETURNS DATETIME
+                READS SQL DATA
+                DETERMINISTIC
+                BEGIN
+                RETURN CONVERT(CONCAT(setDate,' ','01:01:11'), DATETIME);
+                END;";
+
+        $this->db->query($sql);
+    }
+
+    public function testRawSqlUpdateObject()
+    {
+        $data = [];
+
+        $row             = new stdclass();
+        $row->email      = 'derek@world.com';
```

Comment:
Don't use someone's domain name.

```suggestion
        $row->email      = 'derek@example.com';
```

Reply:
These email addresses are in seeder and used for all the tests.

https://github.com/codeigniter4/CodeIgniter4/blob/4.3/tests/_support/Database/Seeds/CITestSeeder.php

Perhaps changing these should be a separate PR?

Reply:
Oh, yes.